src/Host/Client/Impl/BrokerServices/WebServer.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Newtonsoft.Json;
+
+namespace Microsoft.R.Host.Client.BrokerServices {
+    public class WebServer {
```

Comment:
Why is it so complicated?
I'd expect to get a new remote Url that matches local Url, and then broker will handle it. And ideally, it should not be a broker, but the RHost itself.
@int19h should take a look on this PR.